_posts/2018-07-17-RTE-for-automated-kernel-deployment-and-everyday-use.md

```diff
@@ -0,0 +1,369 @@
+---
+post_title: RTE for automated kernel deployment and everyday use
+author: Piotr Król
+layout: post
+published: true
+post_date: 2018-07-17 13:20:00
+
+tags:
+	- ansible
+	- xen
+  - linux
+categories:
+	- Firmware
+	- OS Dev
+---
+
+We continue our effort to enable IOMMU and as side effect I have to play with
+various technologies to exercise reliable development environment which base on
+[RTE](TBD: marketing website).
+
+In this blog post I would like to present semi-automated technique to debug
+firmware, Xen and Linux kernel. The goal is to have set of tools that help in
+enabling various features in Debian-based dom0.
+
+We would like:
+
+* update Linux kernel which is exposed over iPXE server
+* update rootfs served over NFS
+
+I will use following components:
+
+* [PC Engines apu2c](http://pcengines.ch/apu2c2.htm)
+* [RTE](TBD: link to RTE marketing website)
+* [pxe-server](https://github.com/3mdeb/pxe-server) - our dockerized iPXE and
+  NFS server
+* Xen 4.8
+* Linux kernel 4.14.y
+
+My workstation environment is QubesOS 4.0 with Debian stretch VMs, but it should
+not make any difference. I had to workaround one obstacle related to our
+environment, which is behind VPN, but I also wanted to access outside world in
+my fw-dev VM. More information about there can be found [here](https://groups.google.com/d/msg/qubes-users/UakrAG9Frpc/MP9r6XjtAwAJ)
+
+First, I assume that you have working `pxe-server` and [RTE connected to apu2](TBD: blog post based on apu2 theory of operation).
```

Comment:
No link. We currently have no blog post that explains flashing over RTE and robot framework. At the time presentation was made, we still had not open sourced our code

Reply:
@artur-rs when we can have that?